### PR TITLE
GPS test station first unittests

### PR DESCRIPTION
**Pull request by gast04**
ublox is reset, and then measurements are taken of how long it takes to get a first measurement, satellite count check and until the gps location converges

### CODE REVIEW

File: selfdrive/sensord/pigeond.py [hunk 1]
```diff
@@ -116,34 +115,48 @@ def wait_for_backup_restore_status(self, timeout: float = 1.) -> int:
...
+      pigeon.send_with_ack(b"\xb5\x62\x06\x00\x14\x00\x01\x00\x00\x00\xC0\x08\x00\x00\x00\x08\x07\x00\x01\x00\x01\x00\x00\x00\x00\x00\xF4\x80")
+      pigeon.send_with_ack(b"\xb5\x62\x06\x00\x14\x00\x04\xFF\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x1D\x85")
+      pigeon.send_with_ack(b"\xb5\x62\x06\x00\x00\x00\x06\x18")
+      pigeon.send_with_ack(b"\xb5\x62\x06\x00\x01\x00\x01\x08\x22")
+      pigeon.send_with_ack(b"\xb5\x62\x06\x00\x01\x00\x03\x0A\x24")
+
+      # UBX-CFG-RATE (0x06 0x08) measRate: 100(10Hz), navRate: 1, timeRef: 0
       pigeon.send_with_ack(b"\xB5\x62\x06\x08\x06\x00\x64\x00\x01\x00\x00\x00\x79\x10")
+
+      #UBX-CFG-NAV5 (0x06 0x24)
```
Comment (adeebshihadeh) [inline selfdrive/sensord/pigeond.py:145]: be consistent with your style; generally we use a space for comments and no space for commented out code
```suggestion
      # UBX-CFG-NAV5 (0x06 0x24)
```

File: tools/gpstest/README.md [hunk 2]
```diff
@@ -1,38 +1,72 @@
...
+The latest stable version(22.09) does not have the corresponding firmware
 download available at https://downloads.myriadrf.org/project/limesuite. Therefore
 version 20.10 was chosen.
 
+A successful build should gives similar to:
+```
+./LimeUtil --info
+######################################################
+## LimeSuite information summary
+######################################################
```
Comment (adeebshihadeh) [inline tools/gpstest/README.md:37]: I removed a bunch of this stuff from the initial PR, and looks those changes didn't get merged properly.

File: tools/gpstest/inc_ephem_array_size.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/gpssim.h b/gpssim.h
+index c30b227..2ae0802 100644
```
Comment (adeebshihadeh) [inline tools/gpstest/inc_ephem_array_size.patch:2]: put the patches in their own directory

File: tools/gpstest/setup.sh [hunk 4]
```diff
@@ -9,17 +9,25 @@ if [ ! -d LimeSuite ]; then
   cd LimeSuite
   # checkout latest version which has firmware updates available
   git checkout v20.10.0
+  cp ../mcu_error.patch .
+  cp ../reference_print.patch .
```
Comment (adeebshihadeh) [inline tools/gpstest/setup.sh:13]: shouldn't need to move the patches

File: tools/gpstest/setup.sh [hunk 5]
```diff
@@ -9,17 +9,25 @@ if [ ! -d LimeSuite ]; then
...
+  git apply mcu_error.patch
+  git apply reference_print.patch
```
Comment (adeebshihadeh) [inline tools/gpstest/setup.sh:15]: just apply all `patches/*.patch`

File: tools/gpstest/setup.sh [hunk 6]
```diff
@@ -9,17 +9,25 @@ if [ ! -d LimeSuite ]; then
...
 
 if [ ! -d LimeGPS ]; then
   git clone https://github.com/osqzss/LimeGPS.git
   cd LimeGPS
-  sed -i 's/LimeSuite/LimeSuite -I..\/LimeSuite\/src -L..\/LimeSuite\/builddir\/src/' makefile
+
+  cp ../inc_ephem_array_size.patch .
+  cp ../makefile.patch .
+  git apply inc_ephem_array_size.patch
+  git apply makefile.patch
```
Comment (adeebshihadeh) [inline tools/gpstest/setup.sh:29]: same stuff here

File: tools/gpstest/test_gps.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+    pigeon, pm = pd.create_pigeon()
+    pd.init_baudrate(pigeon)
+
+    # deleting the backup does not always work on first try
+    # mostly works on second try
+    got_deleted = False
+    for _ in range(5):
+      # device cold start
+      pigeon.send(b"\xb5\x62\x06\x04\x04\x00\xff\xff\x00\x00\x0c\x5d")
```
Comment (adeebshihadeh) [inline tools/gpstest/test_gps.py:126]: put stuff like this in pigeond

File: tools/gpstest/test_gps.py [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    # receive some messages (restart after cold start takes up to 30seconds)
+    pd.run_receiving(pigeon, pm, 40)
+
+    verify_ubloxgnss_data(ugs)
+    verify_gps_location(gle)
+
+    # skip for now, this might hang for a while
+    #verify_time_to_first_fix(pigeon)
+
+    pd.set_power(False)
```
Comment (adeebshihadeh) [inline tools/gpstest/test_gps.py:160]: move this to a `tearDown()`

File: tools/gpstest/README.md (hunk 2 above)
Comment (gast04) [reply tools/gpstest/README.md]: oh, I misread that

File: tools/gpstest/test_gps.py (hunk 8 above)
Comment (gast04) [reply tools/gpstest/test_gps.py:139]: in v2 there are more tests, and think makes sense to poweroff the device after every test

Comment (adeebshihadeh) [reply tools/gpstest/test_gps.py:136]: yeah, that's what the tearDown does

**Pull request merged**